pkg/controller/windowsmachineconfig/nodeconfig/init.go

```diff
@@ -0,0 +1,38 @@
+package nodeconfig
+
+import (
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+// cache holds the information of the nodeConfig that is invariant for multiple reconciliation cycles. We'll use this
+// information when we don't want to get the information from the global context coming from reconciler
+// but to have something at nodeConfig package locally which will be passed onto other structs. There is no need to
+// invalidate this cache as of now, since the only entry in this workerIgnitionEndPoint which will be immutable. If
+// someone wants to change it, they've to restart the operator which will invalidate the cache automatically.
```

Comment:
"If someone wants to change it, they've to restart the operator which will invalidate the cache automatically"
This needs to be documented elsewhere if we are making this a requirement

Reply:
If the API server endpoint changes underneath from us I doubt it can happen in an non-invasive manner.